ui/components/component-library/popover/popover.stories.tsx

```diff
@@ -0,0 +1,807 @@
+import React, { useState, useEffect } from 'react';
+import { ComponentStory, ComponentMeta } from '@storybook/react';
+import Box from '../../ui/box/box';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  Color,
+  DISPLAY,
+  JustifyContent,
+  TEXT_ALIGN,
+} from '../../../helpers/constants/design-system';
+import { Icon, IconName, IconSize, PopoverHeader, Text } from '..';
+import README from './README.mdx';
+import { Popover, PopoverPosition, PopoverRole } from '.';
+
+export default {
+  title: 'Components/ComponentLibrary/Popover',
+  component: Popover,
+  parameters: {
+    docs: {
+      page: README,
+    },
+  },
+  argTypes: {
+    children: {
+      control: 'text',
+    },
+    position: {
+      options: PopoverPosition,
+      control: 'select',
+    },
+    role: {
+      options: PopoverRole,
+      control: 'select',
+    },
+    className: {
+      control: 'text',
+    },
+  },
+  args: {
+    children: 'Popover',
+  },
+} as ComponentMeta<typeof Popover>;
+
+const Template: ComponentStory<typeof Popover> = (args) => {
+  const [referenceElement, setReferenceElement] = useState();
```

Comment:
I'm more familiar with using `useRef` to set refs. Do we use `useState` here in case the anchor element position updates due to state change or something then the ref would be updated? I think `useRef` won't re-render while `useState` will